File: error-studio-main/audit_content.py
import os
import re

# Route to Component Mapping
routes = {
    "/": "Home.jsx",
    "/photography-agency": "PhotographyAgencyHome.jsx",
    "/digital-agency": "DigitalAgencyHome.jsx",
    "/marketing-agency": "MarketingAgencyHome.jsx",
    "/freelancer-agency": "FreelancerAgencyHome.jsx",
    "/architecture-agency": "ArchitectureAgencyHome.jsx",
    "/creative-solution": "CreativeSolutionHome.jsx",
    "/personal-portfolio": "PersonalPortfolioHome.jsx",
    "/about": "AboutPage.jsx",
    "/service": "ServicesPage.jsx",
    "/service/:serviceDetailsId": "ServiceDetailsPage.jsx",
    "/portfolio": "PortfolioPage.jsx",
    "/portfolio/:portfolioDetailsId": "PortfolioDetailsPage.jsx",
    "/blog": "BlogPage.jsx",
    "/blog/:blogDetailsId": "BlogDetailsPage.jsx",
    "/contact": "ContactPage.jsx",
    "/team": "TeamPage.jsx",
    "/team/:teamDetails": "TeamDetails.jsx",
    "/case-study/:caseStudyDetailsId": "CaseStudyDetailsPage.jsx",
    "/faq": "FaqPage.jsx",
    "/creative-portfolio": "CreativePortfolioHome.jsx",
    "/showcase-portfolio": "ShowcasePortfolioHome.jsx",
    "/case-study-showcase": "CaseStudyShowcaseHome.jsx",
    "/video-showcase": "VideoShowcaseHome.jsx",
    "*": "ErrorPage.jsx"
}

BASE_DIR = r"c:/Users/DIGITAL MARKETING/Downloads/samplASSDD2111_sS/ERROR_STUDIO/src/components/Pages"

def extract_text_from_file(filepath):
    if not os.path.exists(filepath):
        return None

    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Split into sections based on comments
    # Regex for comments: {/* Start ... Section */}
    section_pattern = re.compile(r"\{\/\*\s*Start\s+(.*?)\s+Section\s*\*\/\}(.*?)(\{\/\*\s*End\s+.*?\s+Section\s*\*\/\}|$)", re.DOTALL)
    
    sections = []
    
    # helper to clean and count text
    def process_text_chunk(text_chunk):
        # Extract props like title="...", subtitle="..."
        props_pattern = re.compile(r'\b(title|subtitle|btnText|text|heading|paragraph|label|heroTitle|heroSubtitle)=["\'](.*?)["\']')
        # Extract JSX text content: >Some text<
        jsx_text_pattern = re.compile(r'>([^<>{}]+)<')
        
        # Extract array data (simplistic approach for now)
        # Look for data={...}
        
        matches = []
        
        for m in props_pattern.finditer(text_chunk):
            val = m.group(2).strip()
            if val and not val.startswith("/") and not val.startswith("{"): # Ignore paths and expressions
                matches.append(val)
                
        for m in jsx_text_pattern.finditer(text_chunk):
            val = m.group(1).strip()
            if val:
                matches.append(val)
                
        # Handle some meaningful variable lookups if needed (like funfaceData)
        # For now, we stick to props and direct text for "Zero assumptions" on static content.
        
        return matches

    # Find explicitly marked sections
    found_sections = list(section_pattern.finditer(content))
    
    if found_sections:
        for match in found_sections:
            sec_name = match.group(1).strip()
            sec_content = match.group(2)
            texts = process_text_chunk(sec_content)
            sections.append({
                "name": sec_name,
                "texts": texts
            })
    else:
        # If no sections found, treat whole file as one "Main" section
        texts = process_text_chunk(content)
        sections.append({
            "name": "Main Content",
            "texts": texts
        })
        
    return sections

def generate_sitemap():
    total_project_chars = 0
    total_project_elements = 0
    
    print("Wait... Auditing content...")
    print("")

    for route, filename in routes.items():
        filepath = os.path.join(BASE_DIR, filename)
        sections = extract_text_from_file(filepath)
        
        if sections is None:
            print(f"Error: Could not find file {filename}")
            continue
            
        print(f"ROUTE: {route} ({filename})")
        
        for sec in sections:
            texts = sec["texts"]
            # Filter empty strings just in case
            texts = [t for t in texts if t.strip()]
            
            elem_count = len(texts)
            char_count = sum(len(t) for t in texts)
            preview = f'"{texts[0][:20]}..."' if texts else "No text content"
            
            # Clean up preview (remove HTML tags from preview if any leaked)
            preview = re.sub(r'<[^>]+>', '', preview)
            
            print(f"├── SECTION: {sec['name']}")
            print(f"│   ├── Text Elements: {elem_count}")
            print(f"│   ├── Total Characters: {char_count}")
            print(f"│   └── Content Preview: {preview}")
            
            total_project_chars += char_count
            total_project_elements += elem_count
            
        print("") # Empty line between routes

    print("Total Content Summary")
    print("---------------------")
    print(f"Total Routes: {len(routes)}")
    print(f"Total Text Elements: {total_project_elements}")
    print(f"Total Characters: {total_project_chars}")

if __name__ == "__main__":
    generate_sitemap()

File: audit_content.js
const fs = require("fs");
const path = require("path");

// Route to Page Component Mapping
const routes = {
  "/": "Home.jsx",
  "/photography-agency": "PhotographyAgencyHome.jsx",
  "/digital-agency": "DigitalAgencyHome.jsx",
  "/marketing-agency": "MarketingAgencyHome.jsx",
  "/freelancer-agency": "FreelancerAgencyHome.jsx",
  "/architecture-agency": "ArchitectureAgencyHome.jsx",
  "/creative-solution": "CreativeSolutionHome.jsx",
  "/personal-portfolio": "PersonalPortfolioHome.jsx",
  "/about": "AboutPage.jsx",
  "/service": "ServicesPage.jsx",
  "/service/:serviceDetailsId": "ServiceDetailsPage.jsx",
  "/portfolio": "PortfolioPage.jsx",
  "/portfolio/:portfolioDetailsId": "PortfolioDetailsPage.jsx",
  "/blog": "BlogPage.jsx",
  "/blog/:blogDetailsId": "BlogDetailsPage.jsx",
  "/contact": "ContactPage.jsx",
  "/team": "TeamPage.jsx",
  "/team/:teamDetails": "TeamDetails.jsx",
  "/case-study/:caseStudyDetailsId": "CaseStudyDetailsPage.jsx",
  "/faq": "FaqPage.jsx",
  "/creative-portfolio": "CreativePortfolioHome.jsx",
  "/showcase-portfolio": "ShowcasePortfolioHome.jsx",
  "/case-study-showcase": "CaseStudyShowcaseHome.jsx",
  "/video-showcase": "VideoShowcaseHome.jsx",
  "*": "ErrorPage.jsx",
};

// Component Mapping for recursive lookup
// Maps Component Name -> File Path relative to src/
const componentMapping = {
  Accordion: "components/Accordion/index.jsx",
  TestimonialSlider: "components/Slider/TestimonialSlider.jsx",
  TeamSlider: "components/Slider/TeamSlider.jsx",
  PostSlider: "components/Slider/PostSlider.jsx",
  TimelineSlider: "components/Slider/TimelineSlider.jsx",
  PricingTableList: "components/PricingTable/PricingTableList.jsx",
  FunFact: "components/FunFact/index.jsx",
  LogoList: "components/LogoList/index.jsx",
  // Adding Heroes for completeness if they have static content
  Hero: "components/Hero/index.jsx",
  Hero2: "components/Hero/Hero2.jsx",
  Hero3: "components/Hero/Hero3.jsx",
  Hero4: "components/Hero/Hero4.jsx",
  Hero5: "components/Hero/Hero5.jsx",
  Hero6: "components/Hero/Hero6.jsx",
  Hero7: "components/Hero/Hero7.jsx",
};

// Base dir is now src/ because we need to navigate to both Pages and other components
const BASE_DIR = path.join(__dirname, "src");

function extractTextFromFile(filePathRelative) {
  const filepath = path.join(BASE_DIR, filePathRelative);
  if (!fs.existsSync(filepath)) {
    return null;
  }

  const content = fs.readFileSync(filepath, "utf-8");
  const sections = [];

  // 1. Extract const variables defined in the file
  const varMap = {};
  const varRegex = /const\s+(\w+)\s*=\s*(\[[\s\S]*?\]);/g;
  let match;
  while ((match = varRegex.exec(content)) !== null) {
    varMap[match[1]] = match[2];
  }

  // 2. Scan sections
  const sectionRegex =
    /\{\/\*\s*Start\s+(.*?)\s+Section\s*\*\/\}([\s\S]*?)(\{\/\*\s*End\s+.*?\s+Section\s*\*\/\}|$)/g;

  let foundSections = false;

  while ((match = sectionRegex.exec(content)) !== null) {
    foundSections = true;
    const secName = match[1].trim();
    const secContent = match[2];
    const texts = processTextChunk(secContent, varMap);
    sections.push({
      name: secName,
      texts: texts,
    });
  }

  if (!foundSections) {
    const texts = processTextChunk(content, varMap);
    sections.push({
      name: "Main Content",
      texts: texts,
    });
  }

  return sections;
}

function processTextChunk(chunk, varMap = {}) {
  const matches = [];

  // A. Check for Component usages that need internal content extraction
  for (const [compName, compPath] of Object.entries(componentMapping)) {
    // Regex to find <ComponentName ... /> or <ComponentName>
    const compRegex = new RegExp(`<${compName}\\b`, "g");
    if (compRegex.test(chunk)) {
      // Found a known component, let's extract its content recursively
      // Note: This matches the *usage*, so we load the *definition file* once per usage.
      // This might duplicate content if used multiple times in same section, which is technically correct for a sitemap (it appears twice).

      const compSections = extractTextFromFile(compPath);
      if (compSections) {
        for (const sec of compSections) {
          matches.push(...sec.texts);
        }
      }
    }
  }

  // B. Extract direct props like title="...", subtitle="..."
  const propsRegex =
    /(?:title|subtitle|btnText|text|heading|paragraph|label|heroTitle|heroSubtitle|name|question|answer)=["'](.*?)["']/g;
  let m;
  while ((m = propsRegex.exec(chunk)) !== null) {
    const val = m[1].trim();
    if (val && !val.startsWith("/") && !val.startsWith("{")) {
      matches.push(val);
    }
  }

  // C. Extract props that reference variables: data={funfaceData}
  const dataPropRegex =
    /(?:data|heroSocialLinks|portfolioData|showcaseData|serviceData)=\{(\w+)\}/g;
  while ((m = dataPropRegex.exec(chunk)) !== null) {
    const varName = m[1];
    if (varMap[varName]) {
      const rawData = varMap[varName];
      const objPropRegex =
        /(?:title|subtitle|factNumber|text|btnText|name|testimonialText|question|answer|designation|ratings)\s*:\s*["'](.*?)["']/g;
      let m2;
      while ((m2 = objPropRegex.exec(rawData)) !== null) {
        const val = m2[1].trim();
        if (val && !val.startsWith("/") && !val.startsWith("{")) {
          matches.push(val);
        }
      }
    }
  }

  // D. Extract JSX text content: >Some text<
  const jsxTextRegex = />([^<>{}\n]+)</g;
  while ((m = jsxTextRegex.exec(chunk)) !== null) {
    const val = m[1].trim();
    if (val) {
      matches.push(val);
    }
  }

  return matches;
}

function generateSitemap() {
  let totalProjectChars = 0;
  let totalProjectElements = 0;

  console.log("Wait... Auditing content...\n");

  for (const [route, filename] of Object.entries(routes)) {
    // Pages are in components/Pages
    const sections = extractTextFromFile(
      path.join("components", "Pages", filename),
    );

    if (!sections) {
      console.log(`Error: Could not find file ${filename}`);
      continue;
    }

    console.log(`ROUTE: ${route} (${filename})`);

    let routeTotalChars = 0;
    let routeTotalElements = 0;

    for (const sec of sections) {
      const texts = sec.texts;
      const elemCount = texts.length;
      const charCount = texts.reduce((acc, t) => acc + t.length, 0);

      let preview =
        texts.length > 0
          ? `"${texts[0].substring(0, 20)}..."`
          : "No text content";
      preview = preview.replace(/<[^>]+>/g, "");

      console.log(`├── SECTION: ${sec.name}`);
      console.log(`│   ├── Text Elements: ${elemCount}`);
      console.log(`│   ├── Total Characters: ${charCount}`);
      console.log(`│   └── Content Preview: ${preview}`);

      routeTotalChars += charCount;
      routeTotalElements += elemCount;
    }

    totalProjectChars += routeTotalChars;
    totalProjectElements += routeTotalElements;
    console.log("");
  }

  console.log("Total Content Summary");
  console.log("---------------------");
  console.log(`Total Routes: ${Object.keys(routes).length}`);
  console.log(`Total Text Elements: ${totalProjectElements}`);
  console.log(`Total Characters: ${totalProjectChars}`);
}

generateSitemap();
